internal/config/config.go

```diff
@@ -234,6 +234,20 @@ func (c *AuthConfig) Logout(hostname string) error {
 	return ghConfig.Write(c.cfg)
 }
 
+// IsEnterprise checks if there are any environment variable authentication
+// tokens set for enterprise hosts.
+func (c *AuthConfig) IsEnterprise() bool {
+	// Any non-github.com hostname is fine here
+	dummyHostname := "example.com"
+	var token string
+	if c.tokenOverride != nil {
+		token, _ = c.tokenOverride(dummyHostname)
+	} else {
+		token, _ = ghAuth.TokenFromEnvOrConfig(dummyHostname)
+	}
+	return token != ""
+}
```

Comment:
I think rather than try to contain all this logic inside of `AuthConfig` we are better off doing something simple like:


```suggestion
func (c *AuthConfig) TokenFromEnvOrConfig(hostname string) (string, string) {
	if c.tokenOverride != nil {
	    return c.tokenOverride(hostname)
	}
	return ghAuth.TokenFromEnvOrConfig(hostname)
}
```


That way this can be reused [here](https://github.com/cli/cli/blob/41a457136e3134e5b5e6c17cba2704309ac609bb/pkg/cmd/factory/remote_resolver.go#L89) also.

Reply:
yes, better! Done